### ISSUE TITLE
Failing NewSLO test (rmProg1)

### ISSUE THREAD

**Issue by yuriy0** [hakaru-dev/hakaru#51]
```
Expected result : Weight(1/2/Pi,Bind(Uniform(3,8),a4,Bind(Uniform(1,4),a5,
                           Weight(<W0>,Ret(Pair(a4,a5))))))
 Evaluated result: Weight(1/2/Pi,Bind(Uniform(1,4),a5,Bind(Uniform(3,8),a4,
                           Weight(<W1>,Bind(Gaussian((<W2>,<W3>)^(1/2)),a6,Ret(Pair(a4,a5)))))))
```

Here we get a multidimensional domain (and successfully solve it) but don't manage to evaluate an integral which becomes a value after solution. Before the Domain code, `elim_intsum` would have been called on every integral of that solution to potentially perform such an evaluation, but now `elim_intsum` is called only once, on the top-level integral, after the application of the solution. Domain:-Apply should (optionally) call `do_elim_intsum` (or something like it) on every integral.

The `Bind`s which should be left in the result end up reversed; Domain:-Apply will only try to respect variable order if no interesting solutions are found. Perhaps it should also try to enforce the existing variable order in solutions, when possible.

**Comment by JacquesCarette** [comment]
The flip is not a big deal.  That the integral isn't performed is more of a problem.  Somehow, it feels like things should recurse in ways so that this evaluates - but the details will matter.

**Comment by yuriy0** [comment]
Fixed by de2fb180.
